puzzlesolver/puzzles/bishop.py

```diff
@@ -0,0 +1,289 @@
+from copy import deepcopy
+from . import ServerPuzzle
+from ..util import *
+from ..solvers import IndexSolver
+import math
+
+class Bishop(ServerPuzzle):
+	# For simplicity, only one color of square is considered.
+	puzzleid = 'bishop'
+	author = 'Brian Delaney'
+	puzzle_name = 'Bishop Puzzle'
+	description = """Swap the locations of two sets of bishops on opposite ends of a chessboard, without moving them into threatened positions."""
+	date_created = "October 30, 2020"
+	# Chessboard has y rows, 2x columns
+	variants = {str(i[0]) + "x" + str(i[1]): IndexSolver for i in ((2,5), (2,7), (3,7))}
+	

```

Comment:
Please add another member test_variants as shown in [Hanoi](https://github.com/GamesCrafters/GamesmanPuzzles/blob/cde9fb8ffc3e702529b107557470fca026a2e29d/puzzlesolver/puzzles/hanoi.py#L22). This allows pytest to focus only on a single subset of the puzzles so that the tests don't take too long.